.github/workflows/ci.yml

```diff
@@ -0,0 +1,35 @@
+name: CI
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        ruby:
+          - 2.4
+          - 2.5
+          - 2.6
+          - 2.7
+        gemfile:
+          - Gemfile
+          - gemfiles/activesupport-5.0.gemfile
+          - gemfiles/activesupport-5.1.gemfile
+          - gemfiles/activesupport-6.0.gemfile
+    name: Ruby ${{ matrix.ruby }} ${{ matrix.gemfile }}
+    steps:
+      - uses: actions/checkout@v1
+      - name: Set up Ruby ${{ matrix.ruby }}
+        uses: actions/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+      - name: Set up gemfile
+        run: |
+          export BUNDLE_GEMFILE=${GITHUB_WORKSPACE}/${{ matrix.gemfile }}
```

Comment:
IIRC GitHub Actions does not persist `export`'d environment variables between steps. You can see this in the logs, for example logs for run "Ruby 2.5 gemfiles/activesupport-5.1.gemfile" shows `Installing activesupport 6.1.0`.

I think you'll need to use [`jobs.build.env`](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idenv).

Reply:
Super helpful. Thanks.